agent/src/main/java/org/wso2/testgrid/agent/AgentApplication.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.testgrid.agent;
+
+import com.google.gson.Gson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.agent.beans.OperationRequest;
+import org.wso2.testgrid.agent.beans.OperationResponse;
+import org.wso2.testgrid.agent.listners.OperationResponseListener;
+import org.wso2.testgrid.agent.websocket.ClientEndpoint;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Properties;
+import javax.websocket.CloseReason;
+
+/**
+ * Agent bootstrap class.
+ *
+ * @since 1.0.0
+ */
+public class AgentApplication implements OperationResponseListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(AgentApplication.class);
+    private static final String AGENT_PROP_FILE = File.separator + "opt" + File.separator + "testgrid" +
+            File.separator + "agent-config.properties";
```

Comment:
We'll need to document about this file?

Let's try to use java.nio.Paths next time! :)

Reply:
This will be a automatically generated file by init.sh script which is not required user awareness.

Reply:
Used java.nio.paths with https://github.com/wso2/testgrid/pull/747/commits/b2812b0818ed38457361934f505c931e587efcad